examples_tests/35.TAA/taa.frag

```diff
@@ -19,6 +19,36 @@ float luminanceFromRGB(in vec3 color)
 	return dot(color, vec3(0.299, 0.587, 0.114));
 }
 
+float chebyshevNorm(in vec3 v)
+{
+	vec3 v_ = abs(v);
+	return max(v_.x, max(v_.y, v_.z));
+}
+
+// clips history sample against neighbourhood color space
+vec3 constrainHistSample(in vec3 hist, in vec3 aabb_min, in vec3 aabb_max)
+{
+	if (aabb_min==aabb_max)
+		return aabb_min; //return immediately if all neighbourhood samples were same color
+	// scale needed to transform unit sphere into ellipsoid insribed into the aabb
+	vec3 scale = 0.5*(aabb_max-aabb_min);
+	vec3 inv_scale = vec3(1.0)/scale;
+	vec3 center = 0.5*(aabb_min+aabb_max);
+	
+	// translate into ellipsoid's space and scale hist sample by inverse scale
```

Comment:
for proper ellipsoid calculate 3x3 covariance matrix (use YUV as XYZ for usual Machine Learning/Statistics algorithm)

you can probably do some funky simple unrolled algo since you'll only have 4 samples.

More here:
https://github.com/buildaworldnet/IrrlichtBAW/pull/341/commits/d71102e72144c3d383eb5cc9aa0087d313dca22c#r310388184